app/services/group_stage_service.rb

```diff
@@ -1,32 +1,59 @@
 # frozen_string_literal: true
 
 class GroupStageService
-  def self.generate_group_stage(groups)
-    raise 'Cannot generate group stage without groups' if groups.length.zero?
+  class << self
+    def generate_group_stage(groups)
+      raise 'Cannot generate group stage without groups' if groups.length.zero?
 
-    # raise an error if the average group size is not a whole number
-    raise 'Groups need to be equal size' unless (groups.flatten.length.to_f / groups.length.to_f % 1).zero?
+      # raise an error if the average group size is not a whole number
+      raise 'Groups need to be equal size' unless (groups.flatten.length.to_f / groups.length.to_f % 1).zero?
 
-    groups = groups.map(&method(:get_group_object_from))
-    Stage.new level: -1, groups: groups
-  end
+      groups = groups.map(&method(:get_group_object_from))
+      Stage.new level: -1, groups: groups
+    end
 
-  def self.get_group_object_from(team_array)
-    Group.new matches: generate_all_matches_between(team_array)
-  end
+    def get_group_object_from(team_array)
+      Group.new matches: generate_all_matches_between(team_array),
+                group_scores: team_array.map { |team| GroupScore.new team: team }
+    end
+
+    def generate_all_matches_between(teams)
+      matches = []
+      teams.combination(2).to_a # = matchups
+           .each_with_index do |matchup, i|
+        match = Match.new state: :not_started,
+                          position: i,
+                          match_scores: [
+                            MatchScore.new(team: matchup.first),
+                            MatchScore.new(team: matchup.second)
+                          ]
+        matches << match
+      end
+      matches
+    end
 
-  def self.generate_all_matches_between(teams)
-    matches = []
-    teams.combination(2).to_a # = matchups
-         .each_with_index do |matchup, i|
-      match = Match.new state: :not_started,
-                        position: i,
-                        match_scores: [
-                          MatchScore.new(team: matchup.first),
-                          MatchScore.new(team: matchup.second)
-                        ]
-      matches << match
+    # Updates all group_scores of the given group
+    #
+    # @param group Group the group to update the group_scores in
```

Comment:
in?

Reply:
that is an actual sentence
in german it would be:
"Die gruppe in welcher die group_scores geupdatet werden"

Reply:
too much group in there imo